sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -587,6 +622,8 @@ object StateStore extends Logging {
   /** Stop maintenance thread and reset the maintenance task */
   def stopMaintenanceTask(): Unit = loadedProviders.synchronized {
     if (maintenanceTask != null) {
+      maintenanceThreadPool.stop()
```

Comment:
1. Let's cleanup relevant variables as well, `threadPoolException`, and `maintenancePartitions`. It should be necessary to re-initialize maintenanceTask.
2. For the full safety, I'd have two if statements, to handle the cleanup of `maintenanceThreadPool` and `maintenanceTask` separately.